src/main/java/org/spongepowered/api/command/CommandManager.java

```diff
@@ -109,7 +109,77 @@
      * @throws IllegalArgumentException Thrown if {@code plugin} is not a
      *     plugin instance
      */
-    Optional<CommandMapping> register(Object plugin, CommandCallable callable, List<String> aliases, Function<List<String>, List<String>> callback);
+    Optional<CommandMapping> register(Object plugin, Command callable, List<String> aliases, Function<List<String>, List<String>> callback);
+
+    /**
+     * Register a given command using the given list of aliases.
+     *
+     * <p>If there is a conflict with one of the aliases (i.e. that alias
+     * is already assigned to another command), then the alias will be skipped.
+     * It is possible for there to be no alias to be available out of
+     * the provided list of aliases, which would mean that the command would not
+     * be assigned to any aliases.</p>
+     *
+     * <p>The first non-conflicted alias becomes the "primary alias."</p>
+     *
+     * @param pluginContainer A {@link PluginContainer}
+     * @param callable The command
+     * @param alias An array of aliases
+     * @return The registered command mapping, unless no aliases could be
+     *     registered
+     * @throws IllegalArgumentException Thrown if {@code plugin} is not a
+     *     plugin instance
+     */
+    Optional<CommandMapping> register(PluginContainer pluginContainer, Command callable, String... alias);
+
+    /**
+     * Register a given command using the given list of aliases.
+     *
+     * <p>If there is a conflict with one of the aliases (i.e. that alias
+     * is already assigned to another command), then the alias will be skipped.
+     * It is possible for there to be no alias to be available out of
+     * the provided list of aliases, which would mean that the command would
+     * not be assigned to any aliases.</p>
+     *
+     * <p>The first non-conflicted alias becomes the "primary alias."</p>
+     *
+     * @param pluginContainer A {@link PluginContainer}
+     * @param callable The command
+     * @param aliases A list of aliases
+     * @return The registered command mapping, unless no aliases could be
+     *     registered
+     * @throws IllegalArgumentException Thrown if {@code plugin} is not a
+     *     plugin instance
+     */
+    Optional<CommandMapping> register(PluginContainer pluginContainer, Command callable, List<String> aliases);
+
+    /**
+     * Register a given command using a given list of aliases.
```

Comment:
No, [it explained what it does, not what the use case was](https://github.com/SpongePowered/SpongeAPI/blob/300807f0c87a17a611effb74125d7cf4af595b98/src/main/java/org/spongepowered/api/command/CommandManager.java#L90-L97).

I don't see a use case for this. Plugins should just pass the list of aliases to the function, why should it pass the list and the callback when it can just pass the modified list in the first place?